runtime/v2/builder.go

```diff
@@ -174,7 +174,12 @@ func (a *AppBuilder[T]) Build(opts ...AppBuilderOption[T]) (*App[T], error) {
 			return nil
 		},
 		ExportGenesis: func(ctx context.Context, version uint64) ([]byte, error) {
-			genesisJson, err := a.app.moduleManager.ExportGenesisForModules(ctx)
+			state, err := a.app.db.StateAt(version)
```

Comment:
This feels weird to require that here, this is already done in here: https://github.com/cosmos/cosmos-sdk/blob/561840fa260eb3ba757f83d37fe54091713b1f98/server/v2/appmanager/appmanager.go#L92

If this is needed, IMHO, we should just change `ExportGenesis` arguments from appmanager and pass down the state here.

Reply:
We only need the state for `RunWithCtx`. Since `RunWithCtx` isn't necessary, I've removed fetching the state here and moved it to the runtime builder. This way, we don't have to change the `ExportGenesis` arguments.